src/diffusion_for_multi_scale_molecular_dynamics/score/gaussian_score.py

```diff
@@ -0,0 +1,47 @@
+"""Gaussian Score.
+
+This module implements the Gaussian score, which corresponds to the "perturbation kernel" for diffusion on the lattice
+parameters.
+
+The Gaussian takes the form
+
+    K(l, l0) ~ exp[- |l - l0|^2 / 2 sigma^2]
+
+The score is defined as S = nabla_l ln K(l, l0).
+"""
+
+import torch
+
+
+def get_lattice_sigma_normalized_score(
+    delta_l: torch.Tensor, sigma_n: torch.Tensor, alpha_bar: torch.Tensor
+) -> torch.Tensor:
+    r"""Get the sigma normalized score for lattice parameters.
+
+    We compute this from a normal (non-wrapped) gaussian kernel.  Because we are using a variance-preserving approach
+    for the lattice parameters, the width of the gaussian kernel is rescaled by the noise parameter :math:`alpha_bar`.
+
+    .. math::
+
+        \sigma_\textrm{eff}^2 = (1 - \bar{\alpha}) \sigma^2(n)
+
+    with :math:`\sigma(n) = \frac{\sigma}{\sqrt[1/d]{n}}` with :math:`d` the number of spatial dimensions.
+
+    Args:
+        delta_l : the lattice parameters at which the wrapped Gaussian is evaluated.
+        sigma_n : the variance in the definition of the variance-exploding diffusion - scaled by the number of atoms.
+        alpha_bar : the cumulative variance in the definition of the variance-preserving diffusion.
+
+    Returns:
+        sigma_score : the value of the sigma normalized score.
+    """
+    # rescale sigma by 1 - alpha_bar
+    sigma_effective = torch.sqrt(1 - alpha_bar) * sigma_n
+
+    # we can get the sigma normalized score using a method similar to the brute force calculation in the wrapped
+    # gaussian kernel implementation. We do not need a sum over k as the kernel is not periodic. This simplifies to the
+    # -x / sigma
+
+    sigma_score = -delta_l / sigma_effective
```

Comment:
According to MatterGen, the lattice conditional distribution in the variance preserving scheme looks like this:

<img width="669" alt="Screenshot 2024-12-23 at 4 37 01 PM" src="https://github.com/user-attachments/assets/e7e33480-3a02-42f2-9d99-478c39c859bb" />

The corresponding score will be $\nabla \log q(L_t | L_0)$.

The correct "delta_l$ should thus not be $L-L_0$, but that is what is passed in where this method is called (See my comments in `AXLDiffusionLightningModel`).

I think here is the right place to implement the correct functional form: it would mean changing the signature to pass both $L_t$ and $L_0$, but it would nicely encapsulate this complicated technical detail away from `AXLDiffusionLightningModel`.

Reply:
yes, done